services/users/src/main/java/org/opfab/users/services/UserSettingsService.java

```diff
@@ -176,4 +176,71 @@ public OperationResult<UserSettings> updateUserSettings(String login, UserSettin
 
     }
 
+    public void setProcessStateNotified(String process, String state) {
+        log.info("setProcessStateNotified for {} {}", process, state);
+        List<UserSettings> settings = userSettingsRepository.findAll();
+
+        settings.forEach(userSettings -> {
+            if (userSettings.getProcessesStatesNotNotified() != null && userSettings.getProcessesStatesNotNotified().get(process) != null) {
+                List<String> statesNotNotified = userSettings.getProcessesStatesNotNotified().get(process);
+                int stateIndex = statesNotNotified.indexOf(state);
+                if (stateIndex >= 0) {
+                    statesNotNotified.remove(stateIndex);
+                    userSettingsRepository.save(userSettings);
+                    notificationService.publishUpdatedUserMessage(userSettings.getLogin());
+                }
+            }
+        });
+    }
+
+    public void unsetProcessStateNotified(String process, String state) {
+        log.info("unsetProcessStateNotified for {} {}", process, state);
+        List<UserSettings> settings = userSettingsRepository.findAll();
+
+        settings.forEach(userSettings -> {
+            if (userSettings.getProcessesStatesNotNotified() != null && userSettings.getProcessesStatesNotNotified().get(process) != null) {
```

Comment:
here you add an entry in the repository so even if getProcessStatsNotNotified() is null , you need to add the state as not notified 

The same if userSettings.getProcessesStatesNotNotified().get(process) is null , you need to add it anyway